edpop_explorer/readers/fbtee.py

```diff
@@ -0,0 +1,107 @@
+from typing import Dict, List, Optional, Tuple
+from dataclasses import dataclass, field as dataclass_field
+from pathlib import Path
+import sqlite3
+import yaml
+import requests
+from appdirs import AppDirs
+
+from edpop_explorer.apireader import APIReader, APIRecord, APIException
+
+
+@dataclass
+class FBTEERecord(APIRecord):
+    data: Dict[str, str] = dataclass_field(default_factory=dict)
+    authors: List[Tuple[str, str]] = dataclass_field(default_factory=list)
+
+    def get_title(self) -> str:
+        return self.data.get('full_book_title', '(no title provided)')
+
+    def show_record(self) -> str:
+        return_string = yaml.safe_dump(self.data, allow_unicode=True)
+        if self.link:
+            return_string = self.link + '\n' + return_string
+        if self.authors:
+            authorstrings = [x[0] + ' - ' + x[1] for x in self.authors]
+            return_string += '\nAuthors:\n' + '\n'.join(authorstrings)
+        return return_string
+
+    def __repr__(self):
+        return self.get_title()
+
+
+class FBTEEReader(APIReader):
+    DATABASE_URL = 'https://dhstatic.hum.uu.nl/edpop/cl.sqlite3'
+    DATABASE_LICENSE = 'https://dhstatic.hum.uu.nl/edpop/LICENSE.txt'
+    FBTEE_LINK = 'http://fbtee.uws.edu.au/stn/interface/browse.php?t=book&' \
+        'id={}'
+
+    def __init__(self):
+        self.database_file = Path(
+            AppDirs('edpop-explorer', 'cdh').user_data_dir
+        ) / 'cl.sqlite3'
+        if not self.database_file.exists():
+            self._download_database()
+        self.con = sqlite3.connect(str(self.database_file))
+
+    def _download_database(self):
+        print('Downloading database...')
+        response = requests.get(self.DATABASE_URL)
+        if response.ok:
+            try:
+                with open(self.database_file, 'wb') as f:
+                    f.write(response.content)
+            except OSError as err:
+                raise APIException(
+                    'Error writing database file to disk: {}'.format(err)
+                )
+        else:
+            raise APIException(
+                'Error downloading database file from {}'
+                .format(self.DATABASE_URL)
+            )
+        print('Successfully saved database to {}.'.format(self.database_file))
+        print('See license: {}'.format(self.DATABASE_LICENSE))
+
+    def prepare_query(self, query: str):
+        self.prepared_query = '%' + query + '%'
+
+    def fetch(self) -> List[FBTEERecord]:
+        if not self.prepared_query:
+            raise APIException('First call prepare_query method')
+
+        cur = self.con.cursor()
+        columns = [x[1] for x in cur.execute('PRAGMA table_info(books)')]
+        res = cur.execute(
+            'SELECT B.*, BA.author_code, A.author_name FROM books B '
+            'JOIN books_authors BA on B.book_code=BA.book_code '
```

Comment:
Use a `LEFT JOIN`, since if there is no author to be joined, a book record will be exluded (example: `spbk0000067`).